node/pkg/watchers/evm/verify_chain_config/verify.go

```diff
@@ -132,3 +152,27 @@ func verifyFinality(ctx context.Context, url string, finalized, safe bool) error
 
 	return nil
 }
+
+func verifyContractAddr(ctx context.Context, url string, contractAddr string) error {
+	timeout, cancel := context.WithTimeout(ctx, 15*time.Second)
+	defer cancel()
+
+	rawClient, err := ethRpc.DialContext(timeout, url)
+	if err != nil {
+		return fmt.Errorf("failed to connect: %w", err)
+	}
+
+	client := ethClient.NewClient(rawClient)
+
+	caller, err := ethAbi.NewAbiCaller(ethCommon.BytesToAddress(ethCommon.HexToAddress(contractAddr).Bytes()), client)
+	if err != nil {
+		return fmt.Errorf("failed to create abi caller: %w", err)
+	}
+
+	_, err = caller.GetCurrentGuardianSetIndex(&ethBind.CallOpts{Context: timeout})
```

Comment:
should the result here be checked?

Reply:
We're just trying to confirm that the contract is at that address, so I didn't think it was necessary. And since it  returns a `uint32`, I don't even know what I would check for.